File: ru.kmorozov.gbd.core.logger/src/main/java/ru/kmorozov/gbd/logger/progress/IProgress.java
package ru.kmorozov.gbd.logger.progress;

/**
 * Created by km on 06.11.2016.
 */
public interface IProgress {

    int inc();

    int incrementAndProgress();

    void finish();

    IProgress getSubProgress(int maxValue);

    void resetMaxValue(int maxValue);
}

File: ru.kmorozov.gbd.core.logger/src/main/java/ru/kmorozov/gbd/logger/progress/IProgress.kt
package ru.kmorozov.gbd.logger.progress

/**
 * Created by km on 06.11.2016.
 */
interface IProgress {

    fun inc(): Int

    fun incrementAndProgress(): Int

    fun finish()

    fun getSubProgress(maxValue: Int): IProgress

    fun resetMaxValue(maxValue: Int)
}
